interface/forms/group_attendance/functions.php

```diff
@@ -0,0 +1,214 @@
+        $sqlBindArray = array();
+        array_push($sqlBindArray, $pc_aid, $pid, $gid, $pc_eventDate, $recurrspec, $pc_startTime, $participantData['status']);
+        sqlInsert($insert_sql, $sqlBindArray);
+    }
+}
+
+/**
+ * Creates an encounter for patient from attendance form
+ * @param $pid
+ * @param $gid
+ * @param $group_encounter_date
+ * @param $participantData
+ * @param $pc_aid
+ */
+function insert_patient_encounter($pid, $gid, $group_encounter_date, $participantData, $pc_aid){
+    $select_sql = "SELECT id FROM form_encounter WHERE pid = ? AND external_id = ? AND pc_catid = ? AND date = ?; ";
+    $result = sqlStatement($select_sql, array($pid, $gid, 1000, $group_encounter_date));
+    $result_array = sqlFetchArray($result);
+    if($result_array){
+        $insert_sql = "UPDATE form_encounter SET reason = ? WHERE id = ?;";
+        sqlInsert($insert_sql, array($participantData['comment'], $result_array['id']));
+    }
+    else{
+        $insert_encounter_sql =
+            "INSERT INTO form_encounter (date, reason, pid, encounter, pc_catid, provider_id, external_id) ".
+            "VALUES (?, ?, ?, ?, 1000, ?, ?);";
+        $enc_id = generate_id();
+        $sqlBindArray = array();
+        array_push($sqlBindArray, $group_encounter_date, $participantData['comment'], $pid, $enc_id, $pc_aid, $gid);
+        $form_id = sqlInsert($insert_encounter_sql, $sqlBindArray);
+
+        global $userauthorized;
+
+        addForm($enc_id, "New Patient Encounter", $form_id, "newpatient",$pid, $userauthorized, $group_encounter_date,'','',NULL);
+
+    }
+}
+
+/**
+ * If the group encounter was created in relation to a group appointment, fetches the appointment relevant data.
+ * @param $encounter_id
+ * @return array
+ */
+function get_appt_data($encounter_id){
+    $sql =
+        "SELECT ope.pc_aid, ope.pc_eventDate, ope.pc_startTime FROM form_groups_encounter as fge " .
+        "JOIN openemr_postcalendar_events as ope ON fge.appt_id = ope.pc_eid " .
+        "WHERE fge.encounter = ?;";
+    $result = sqlStatement($sql, array($encounter_id));
+    $result_array = sqlFetchArray($result);
+    return $result_array;
+}
+
+/**
+ * Gets group encounter data
+ * @param $encounter_id
+ * @return array
+ */
+function get_group_encounter_data($encounter_id){
+    $sql = "SELECT date FROM form_groups_encounter WHERE encounter = ?";
+    $result = sqlStatement($sql, array($encounter_id));
+    $result_array = sqlFetchArray($result);
+    return $result_array;
+}
+
+/**
+ * Checks if to create appointment and encounter for patient himself based on the status in the attendance form.
+ * @param $status
+ * @return bool
+ */
+function if_to_create_for_patient($status){
+    if($status == '@' || $status == '?' || $status == '~')
```

Comment:
As an aside, are these the same statuses for appt, or another status list?
It is better to not hard-code these things. Check out the mechanism here which is set in the administration->lists settings for the check in statuses(I am assuming these are check in statuses):
https://github.com/matrix-israel/openemr/blob/4a6016bb0537b2d498a6298b2b71ac7b6aa5ecff/library/patient_tracker.inc.php#L157-L164